### PR TITLE
GH408: Adds async example to documentation

### LINKED ISSUE #408: Document Async Cake post 0.23.0 release
cake-build/cake#1258 added support for async Tasks, this should be documented.

### PR DESCRIPTION
**Pull request by devlead**
Please review but don't merge until `0.23.0` is released.
* fixes #408

### CODE REVIEW

**Comment by devlead** [comment]
@cake-build/cake-team 0.23.0 has now shipped so this can be merged and perhaps even referred to in blog post.

File: input/docs/fundamentals/asynchronous-tasks.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+Order: 55
+---
+
+# Asynchronous Tasks
+
+Cake supports having asynchronous tasks since version 0.23.0, this will let you use the `async` and `await` C# features within a task.
+
+This is really simplifies asynchronous programming within your Cake scripts.
```
Comment (gep13) [inline input/docs/fundamentals/asynchronous-tasks.md]: I think this needs to change...

`This really simplifies...`

Comment (devlead) [reply input/docs/fundamentals/asynchronous-tasks.md]: to what? :)
Perhaps:
`This makes it easier to do asynchronous programming within your Cake scripts.`
?

Comment (gep13) [reply input/docs/fundamentals/asynchronous-tasks.md]: You currently have `This is really simplifies...` and I was suggesting that you change it to `This really simplifies...`  i.e. remove the `is` as that part doesn't really make sense.

Comment (devlead) [reply input/docs/fundamentals/asynchronous-tasks.md]: Ah sorry :) I'll fix.

**Comment by devlead** [comment]
@gep13 addressed feedback.

**Comment by devlead** [comment]
@gep13 added infoabout that task execution order & parallelism  isn't affected by this.

**Comment by gep13** [comment]
@devlead your changes have been merged, thanks for your contribution 👍

**Pull request merged**